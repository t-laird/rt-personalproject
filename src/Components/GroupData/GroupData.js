import React, { Component } from 'react';
import './GroupData.css';
import { connect } from 'react-redux';

export class GroupData extends Component {

  totalPoints = () => {
    const totals = this.props.GroupTransactions.reduce((accum, week) => {
      accum += week.transactions.reduce((accum, transaction) => {
        accum += transaction.point_value;
        return accum;
      }, 0);
      return accum;
    }, 0);
    return totals;
  }

	currentPoints = () => {
	  if (this.props.GroupTransactions.length > 0) {
	    const allCurrent = (this.props.GroupTransactions[this.props.GroupTransactions.length - 1]);			
	    const currentTotals = allCurrent.transactions.reduce((accum, transaction) => {
	      accum += transaction.point_value;
	      return accum;
	    }, 0);

	    return currentTotals;
	  }
	}

	render() {
	  return (
	    <div className="group-data-component">
	      <h2>{this.props.group.group_name || 'Group'}</h2>
	      <h3>Group points awarded this week: <span className="group-span">{this.currentPoints()}</span></h3>
	      <h3>Total - all time: <span className="group-span">{this.totalPoints()}</span></h3>
	    </div>
	  ); 
	}
}

export const mapStateToProps = ( store ) => ({
  group: store.Group,
  GroupTransactions: store.GroupTransactions
});

export default connect(mapStateToProps, null)(GroupData);