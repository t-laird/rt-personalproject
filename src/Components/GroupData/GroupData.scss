@import '../../Variables.scss';

.group-data-component {
	background-color: $c3;
	@include center-box;

	h4 {
		font-weight: 900;
		color: $c7;
		text-align: center;
	}

	.group-header {
		@include box-header;
		background-color: $c2;
		width: 100%;
	}

	h3 {
		margin: 0 0 13px 20px;
	}

	.current-points {
	  @include center-box;
	  background: $c3;
	  padding: 14px 25px 8px;
	  display: flex;
	  justify-content: center;

	  div {
	    margin: 20px 0px;
	    span.group-span {
	      display: flex;
	      justify-content: center;
	      align-items: center;
	      font-size: 60px;
	      font-family: "semplicitapro", sans-serif;
	      font-weight: 900;
	      height: 130px;
	      width: 224px;
	      background: $c4;
	      color: $c7;
	      border-bottom: 11px solid $c7;
	      margin-bottom: 3px;
	    }
	    h6 {
	      color: $c4;
	      text-align: center;
	      font-weight: 900;
	    }
	  }

	  div.divider {
	    display: flex;
	    flex-direction: column;
	    justify-content: center;
	    width: 170px;
	    h4 {
	      width: 100%;
	      color: $c4;
	      font-family: "semplicitapro", sans-serif;
	      font-weight: 900;
	      text-align: center; 
	      font-size: 34px;
	    }
	    span {
	      width: 100%;
	      color: $c7;
	      font-size: 23px;
	      font-weight: 900;
	      font-family: "semplicitapro", sans-serif;   
	      text-align: center;
	      padding-bottom: 8px;  
	    }
	  }
	}
}

@media screen and (max-width: 750px) {
	.group-data-component {
		@include media-width;

		.group-header, .current-points {
			@include media-width
		}

		.current-points {
			flex-direction: column;
			align-items: center;

			.divider {
				order: -1;
			}
		}

		.current-points div {
			margin: 10px 0;
		}
	}
}
