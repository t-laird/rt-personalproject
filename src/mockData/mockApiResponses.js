export const getGroupSettingsApiResponse = {
  status: 'success',
};

export const getGroupTransactionDataApiResponse = {
	status: 'success'
};