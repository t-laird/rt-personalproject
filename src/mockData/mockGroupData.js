export const mockGroupData = {
	administrator_id: 95,
	created_date: "2017-10-18T18:34:30.017Z",
	group_id: 1,
	group_name: 'Turing',
	group_passphrase: "9139z",
	weekly_points: 100
}