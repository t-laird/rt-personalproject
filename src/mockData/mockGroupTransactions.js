const mockGroupTransactions = [
  {transactions: [
    {
      created_time: 1514939019586,
      event_id: 9,
      event_time: "2018-01-03T00:23:39.627Z",
      group_id: 1,
      note: "You totally killed it on your lightning talk this morning!  I loved it.",
      point_value: 15,
      receive_id: 22,
      received_name: "Scott Adams",
      send_id: 24,
      send_name: "Jim Davis"
    },
    {
      created_time: 1514939019586,
      event_id: 2,
      event_time: "2018-01-03T00:23:39.612Z",
      group_id: 1,
      note: "Thanks!",
      point_value: 10,
      receive_id: 59,
      received_name: "Bo Jackson",
      send_id: 43,
      send_name: "Deion Sanders"
    }
  ]}
]

export default mockGroupTransactions;