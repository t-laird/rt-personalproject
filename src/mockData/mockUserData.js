export const mockUserData = {
  authrocket_id: "usr_0vXWX9jSRkbUjLGsOb9S8r",
  created_date: "2017-11-18T18:34:30.017Z",
  email: "cpavin@usopen.com",
  group_id: 1,
  name: "Corey Pavin",
  user_id: 95
};

