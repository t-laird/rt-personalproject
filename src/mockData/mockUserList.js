export const mockUserList = [
  {
    authrocket_id: "jfnxcio42393",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "abcf@gmail.com",
    group_id: 1,
    name: "Mike Jenkins",
    user_id: 2
  },
  {
    authrocket_id: "lksdfjn23413",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "qwi123@gmail.com",
    group_id: 1,
    name: "Josh Brown",
    user_id: 3
  },
  {
    authrocket_id: "xmcv3941nv",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "293xcnv@gmail.com",
    group_id: 1,
    name: "Alan Michaelson",
    user_id: 4
  },
  {
    authrocket_id: "3490xxa1j",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "nc-1n23a@gmail.com",
    group_id: 1,
    name: "Sarah Hilman",
    user_id: 5    
  } 
];