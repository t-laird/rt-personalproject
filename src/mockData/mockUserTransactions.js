const mockUserTransactions = [
  {
    received: [
      {
        'send_id': 5,
        'receive_id': 8,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 1,
        'receive_id': 8,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ],
    sent: [
      {
        'send_id': 8,
        'receive_id': 4,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 8,
        'receive_id': 3,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ]
  }
];

export default mockUserTransactions;

