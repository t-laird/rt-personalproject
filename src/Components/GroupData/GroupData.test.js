import React from 'react';
import { GroupData, mapStateToProps } from './GroupData';
import { mockGroupData } from '../../mockData/mockGroupData';
import mockGroupTransactions from '../../mockData/mockGroupTransactions';
import { shallow } from 'enzyme';

describe('group data tests', () => {
  it('should match the snapshot', () => {
    const renderedApp = shallow(<GroupData group={mockGroupData} GroupTransactions={mockGroupTransactions} />);

    expect(renderedApp).toMatchSnapshot();
  })

  it('total should return a total of all group points', () => {
    const renderedApp = shallow(<GroupData group={mockGroupData} GroupTransactions={mockGroupTransactions} />);

    expect(renderedApp.instance().totalPoints()).toEqual(25);
  })

  it('currentPoints should return a total of current group points', () => {
    const renderedApp = shallow(<GroupData group={mockGroupData} GroupTransactions={mockGroupTransactions} />);

    expect(renderedApp.instance().currentPoints()).toEqual(25);
  })
})

describe('mapStateToProps tests', () => {
  it('should pull group transactions from store', () => {
    const mockStore = { mockGroupTransactions };
    const result = mapStateToProps(mockStore);

    expect(result.GroupTransactions).toEqual(mockStore.GroupTransactions);
  })

  it('should pull group data from store', () => {
    const mockStore = { mockGroupData };
    const result = mapStateToProps(mockStore);

    expect(result.group).toEqual(mockStore.group);
  })
})
