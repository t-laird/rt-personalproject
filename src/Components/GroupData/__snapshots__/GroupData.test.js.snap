// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group data tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupData
    GroupTransactions={
        Array [
            Object {
              "transactions": Array [
                Object {
                  "created_time": 1514939019586,
                  "event_id": 9,
                  "event_time": "2018-01-03T00:23:39.627Z",
                  "group_id": 1,
                  "note": "You totally killed it on your lightning talk this morning!  I loved it.",
                  "point_value": 15,
                  "receive_id": 22,
                  "received_name": "Scott Adams",
                  "send_id": 24,
                  "send_name": "Jim Davis",
                },
                Object {
                  "created_time": 1514939019586,
                  "event_id": 2,
                  "event_time": "2018-01-03T00:23:39.612Z",
                  "group_id": 1,
                  "note": "Thanks!",
                  "point_value": 10,
                  "receive_id": 59,
                  "received_name": "Bo Jackson",
                  "send_id": 43,
                  "send_name": "Deion Sanders",
                },
              ],
            },
          ]
    }
    group={
        Object {
            "administrator_id": 95,
            "created_date": "2017-10-18T18:34:30.017Z",
            "group_id": 1,
            "group_name": "Turing",
            "group_passphrase": "9139z",
            "weekly_points": 100,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Turing
</h2>,
        <h3>
          Group points awarded this week: 
          <span
                    className="group-span"
          >
                    25
          </span>
</h3>,
        <h3>
          Total - all time: 
          <span
                    className="group-span"
          >
                    25
          </span>
</h3>,
      ],
      "className": "group-data-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Turing",
        },
        "ref": null,
        "rendered": "Turing",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Group points awarded this week: ",
            <span
              className="group-span"
>
              25
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Group points awarded this week: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 25,
              "className": "group-span",
            },
            "ref": null,
            "rendered": 25,
            "type": "span",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Total - all time: ",
            <span
              className="group-span"
>
              25
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Total - all time: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 25,
              "className": "group-span",
            },
            "ref": null,
            "rendered": 25,
            "type": "span",
          },
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Turing
</h2>,
          <h3>
            Group points awarded this week: 
            <span
                        className="group-span"
            >
                        25
            </span>
</h3>,
          <h3>
            Total - all time: 
            <span
                        className="group-span"
            >
                        25
            </span>
</h3>,
        ],
        "className": "group-data-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Turing",
          },
          "ref": null,
          "rendered": "Turing",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Group points awarded this week: ",
              <span
                className="group-span"
>
                25
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Group points awarded this week: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 25,
                "className": "group-span",
              },
              "ref": null,
              "rendered": 25,
              "type": "span",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Total - all time: ",
              <span
                className="group-span"
>
                25
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Total - all time: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 25,
                "className": "group-span",
              },
              "ref": null,
              "rendered": 25,
              "type": "span",
            },
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
